### PR TITLE
Notebooks update

### PR DESCRIPTION
**Pull request by shaydeci**
Segmentation: transfer learning, quickstart and custom dataset connection.
Updated readme links + makefile

### CODE REVIEW

**Review by Louis-Dupont — COMMENTED** [review]
I didnt manage to comment directly on  the notebooks, so here are a few notes

This cell doesnt print anything. Is it on purpose (cleared) or unintended ?
``` python
from prettyformatter import pprint

print('Dataloader parameters:')
pprint(train_loader.dataloader_params)
print('Dataset parameters')
pprint(train_loader.dataset.dataset_params)
```

*Note: I just saw that in `transfer_learning_semantic_segmentation.ipynb`  is it shown but in the next cell 
image


We are not using infery, so lett's remove it
```
! pip install -qq infery
```

File: README.md [hunk 1]
```diff
@@ -211,9 +211,10 @@ model = models.get("model-name", pretrained_weights="pretrained-model-name")
...
-* [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://bit.ly/3qKwMbe) [Segmentation Transfer Learning](https://bit.ly/3qKwMbe)
-* [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://bit.ly/3QQBVJp) [How to Connect Custom Dataset](https://bit.ly/3QQBVJp)
+* [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Deci-AI/super-gradients/blob/master/notebooks/quickstart_segmentation.ipynb) [Segmentation Quick Start](https://bit.ly/3qKx9m8)
+* [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Deci-AI/super-gradients/blob/master/notebooks/transfer_learning_semantic_segmentation.ipynb) [Segmentation Transfer Learning](https://bit.ly/3qKx9m8)
+* [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Deci-AI/super-gradients/blob/master/notebooks/segmentation_connect_custom_dataset.ipynb) [How to Connect Custom Dataset](https://bit.ly/3qKx9m8)
+
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply README.md]: fixed 5d2c93fffeed71ccd55f7a54e6c1c1866d6f9346

**Comment by shaydeci** [comment]
> I didnt manage to comment directly on the notebooks, so here are a few notes
> 
> 1. This cell doesnt print anything. Is it on purpose (cleared) or unintended ?
> 
> ```python
> from prettyformatter import pprint
> 
> print('Dataloader parameters:')
> pprint(train_loader.dataloader_params)
> print('Dataset parameters')
> pprint(train_loader.dataset.dataset_params)
> ```
> 
> *Note: I just saw that in `transfer_learning_semantic_segmentation.ipynb` is it shown but in the next cell image
> 
> 2. We are not using infery, so lett's remove it
> 
> ```
> ! pip install -qq infery
> ```

Removed infery refs in 5d2c93fffeed71ccd55f7a54e6c1c1866d6f9346

**Comment by BloodAxe** [comment]
Few notes on my end:
1) prettyformatter is not needed and we can remove it, since there is `pprint` package with is supported in all python 3.x versions and does the same stuff as prettyformatter 

2) Printing to stdout is broken in our notebooks, when using DataLoaders, I have a fix for that which I can do a

3) Quickstart segmentation notebook uses ema setting without explicitly specifying `decay_type`, which I think we should change to silence the warning


4) Quickstart segmentation notebook installs `onnx-simplifier` without any version specifier, which is a) bad practice b) unnecessary since we have `onnx-simplifier` already part of SG dependencies. (See section 7) I suggest removing this install.

4) Quickstart segmentation notebook installs torch 1.12 (See section 7) without saying a word why we ask for this specific version. I suggest removing this install as well since it's probably comes from very very old times.

**Comment by BloodAxe** [comment]
6. Export call uses opset_version=11 which is pretty old, suggest to remove this explicit opset argument at all.

**Comment by BloodAxe** [comment]
# segmentation_connect_custom_dataset.ipynb
1. Same with prettyformatter
2. Strange naming of sections `2.B Create Torch Dataset` & `2.C Create Torch Dataloader`. Why not 2.1 / 2.2? My first take was this some kind of typo
3. Notebook emits warning `DeprecationWarning: Object name `cross_entropy` is now deprecated. Please replace it with `CrossEntropyLoss`.`

**Comment by BloodAxe** [comment]
## notebooks/transfer_learning_semantic_segmentation.ipynb

1) prettyformatter
2) `SUPERVISELY_DATASET_DOWNLOAD_PATH="/home/data"` has two issues. First - is requires Linux/Mac OS, second - it assumes that /home is writable which may not be the case (Since it's home folder for all users and not the current one). Let's avoid using absolute directories and use `os.cwd()` or simply `./` and build our path from it instead. We want to follow the rule that notebook should be able to be executed "as is" without any changes in cells.

3) `supervisely_dataset_dir_path` should be constructed using `os.path.join` or `pathlib.Path` and not via string concatenation. It's just a bad practice.

4) Notebook emits `DeprecationWarning: Object name `bce_dice_loss` is now deprecated. Please replace it with `BCEDiceLoss`.`

5) Notebook emits `WARNING - ema.py - Parameter `decay` is not specified for EMA params. Please specify `decay` parameter explicitly in your config`

**Review by BloodAxe — APPROVED** [review]

**Pull request merged**